internal/devbox/secrets.go

```diff
@@ -9,41 +9,36 @@ import (
 	"go.jetpack.io/pkg/envvar"
 )
 
-type secrets struct {
-	envsec.Envsec
-	EnvName string
-}
-
-func (d *Devbox) Secrets(ctx context.Context) (*secrets, error) {
-	envsecInstance := envsec.Envsec{
+func (d *Devbox) UninitializedSecrets(ctx context.Context) *envsec.Envsec {
+	return &envsec.Envsec{
 		APIHost: build.JetpackAPIHost(),
 		Auth: envsec.AuthConfig{
 			ClientID: envvar.Get("ENVSEC_CLIENT_ID", build.ClientID()),
 			Issuer:   envvar.Get("ENVSEC_ISSUER", build.Issuer()),
 		},
 		IsDev:      build.IsDev,
 		Stderr:     d.stderr,
+		Store:      &jetstore.JetpackAPIStore{},
 		WorkingDir: d.ProjectDir(),
+		EnvID: envsec.EnvID{
+			EnvName: d.environment,
+		},
 	}
+}
+
+func (d *Devbox) Secrets(ctx context.Context) (*envsec.Envsec, error) {
+	envsecInstance := d.UninitializedSecrets(ctx)
 
-	store := &jetstore.JetpackAPIStore{}
-	if err := envsecInstance.SetStore(ctx, store); err != nil {
+	project, err := envsecInstance.ProjectConfig()
+	if err != nil {
 		return nil, err
 	}
-	return &secrets{
-		Envsec:  envsecInstance,
-		EnvName: d.environment,
-	}, nil
-}
+	envsecInstance.EnvID.ProjectID = project.ProjectID.String()
+	envsecInstance.EnvID.OrgID = project.OrgID.String()
```

Comment:
its okay for now, but I feel the API could be made a bit safer by initializing `EnvID` as a complete struct in one place...

having it split between UninitializedSecrets and here feels like it could lead to bugs over time (once someone less experienced with this code refactors it, etc.)

Reply:
yeah, this weirdness is to fix a bug that `secrets init` would not work because it depended on the project already existing. Agree it's a bit awkward API. 

2 possible fixes:

* We can always set the full EnvID object. Re-writing is fine.
* The `UninitializedSecrets` doesn't really need EnvID set. It just felt weird not to set it.